session/manager.go

```diff
@@ -44,30 +62,58 @@ func (m *Manager) Driver(name ...string) (sessioncontract.Driver, error) {
 	}
 
 	if m.drivers[driverName] == nil {
-		if m.customDrivers[driverName] == nil {
-			return nil, fmt.Errorf("driver [%s] not supported", driverName)
-		}
-
-		m.drivers[driverName] = m.customDrivers[driverName]
+		return nil, fmt.Errorf("driver [%s] not supported", driverName)
 	}
 
 	return m.drivers[driverName], nil
 }
 
-func (m *Manager) Extend(driver string, handler func() sessioncontract.Driver) sessioncontract.Manager {
-	m.customDrivers[driver] = handler()
-	return m
+func (m *Manager) Extend(driver string, handler sessioncontract.Driver) (sessioncontract.Manager, error) {
```

Comment:
We shouldn't change it. It was intentionally of the `func()` type so that the user can perform some tasks when registering it.